Toggle (#17)
Royal-Navy/standards-toolkit: packages/documentation/library/pages/develop/components/toggle.md

---
title: Toggle
description: A switch like component with a true or false state, can be used in forms or standalone.
audience: public
pageClasses: ''
context: 'Component Library'
template: withsidebar
---

# Overview

A toggle is a simple component that acts like a switch, with a tue or false state, though when used in HTML forms this changes and results in the inclusion or exclusion of a value.

## Usage

The toggle component can be used as a Vue/React component anywhere in a page or within a html form.

### HTML

> When used in a regular web page the toggle component can only be used in a HTML form and uses markup identical to a checkbox but with classes to indicate the component should be progressivey enhanced to appear like a toggle.

Reviewer: ```suggestion
The HTML version of the component uses a checkbox input to toggle its state. The `.rn-toggle` classes progressively enhance the look of the checkbox into that of the toggle switch.
```

#### Example

![Toggle example](images/toggle.png)

```
<label class="rn-toggle" label-for="happy">
  <input
    class="rn-toggle__input"
    name="happy"
    type="checkbox"
    value="yes"
  />
  <span class="rn-toggle__label-inner">Toggle title</span>
</label>
```

### Vue JS

> A Vue JS toggle can be used either in a form or on it's own, maybe to control part of a page.

Reviewer: ```suggestion
A Vue JS toggle can be used either in a form or on its own.
```
